blog post AWS Security Lake (#850)
falcosecurity/falco-website: content/en/blog/falco-aws-security-lake/index.md

- `checkidentity`: Falcosidekick will try its accesses to AWS at bootstrap, if it fails, all AWS outputs are disabled
- `securitylake.bucket`: the bucket name returned by AWS Security Lake at Custom source creation (mandatory to enable the output)
- `securitylake.region`: the region of the events, as a single bucket can store events from different regions (mandatory)
- `securitylake.accountid`: the accountId of the account where the events have been created, as a single bucket can store events from different accounts (mandatory)
- `securitylake.interval`: time in minutes between 2 PUTs of parquet files into S3 to follow the service's requirements
- `securitylake.batchsize`: max number of events by parquet file, to avoid to reach max allowed size
- `securitylake.minimumpriority`: filter the events we send to AWS Security Lake by their priority

## Deployment of Falco & Falcosidekick

These classic Helm commands will deploy everything for you:

```shell
helm repo add falcosecurity https://falcosecurity.github.io/charts
helm repo update
helm install falco -n falco -f values.yaml --create-namespace
```

> Falco and Falcosidekick will be installed and you should get in the Falcosidekick pods the following logs:

Reviewer: ```suggestion
The Falco and Falcosidekick configurations are almost ready, the next step is getting the following logs into the Falcosidekick pods:
```

```
2023/03/13 13:03:06 [INFO]  : Enabled Outputs : [AWSSecurityLake]
2023/03/13 13:03:06 [INFO]  : Falco Sidekick is up and listening on :2801
```

> Falcosidekick is ready to receive the events and put them to AWS Security Lake!

Reviewer: ```suggestion
Falcosidekick is ready to receive events and upload them into AWS Security Lake!
```

## Results

> ### Logs of Falcosidekick

Reviewer: ```suggestion
### Falcosidekick logs
```

> When Falco will trigger some of its rules, Falcosidekick will receive and queue them for 5 minutes, then it will create a parquet file and put into the S3 bucket:

Reviewer: ```suggestion
When Falco triggers any rules, Falcosidekick will receive and queue them every 5 minutes, then it will create a parquet file and load them into the S3 bucket:
```

2023/03/13 13:30:56 [INFO]  : AWS SecurityLake. - Event queued (b51449f4-bb2d-4ce4-afd8-397bf28b3345)
2023/03/13 13:31:17 [INFO]  : AWS SecurityLake. - Event queued (8a51e8f2-9b78-4aa1-a9e6-55f304c6bea6)
2023/03/13 13:31:37 [INFO]  : AWS SecurityLake. - Event queued (38985e4a-be74-4033-b807-724caac55f7b)
2023/03/13 13:31:41 [INFO]  : AWS SecurityLake. - Event queued (ad8e0bc7-e0c2-4583-9c10-561786268355)
2023/03/13 13:31:42 [INFO]  : AWS SecurityLake. - Event queued (72a002f0-9460-4fe9-af6a-d05f35d65707)
2023/03/13 13:32:00 [INFO]  : AWS SecurityLake. - Event queued (4986a846-c086-4d06-8b10-eec31132a465)
2023/03/13 13:32:28 [INFO]  : AWS SecurityLake. - Event queued (1000b0fe-76d2-4eba-b7cc-84e1bf0f8a53)
2023/03/13 13:32:50 [INFO]  : AWS SecurityLake. - Event queued (ec9c4cd3-d705-4bd4-bb51-4a7cb4ee4b4c)
2023/03/13 13:33:15 [INFO]  : AWS SecurityLake. - Upload parquet file d6b40894-f53c-46b3-9f85-67a7fbbdd56b.parquet OK (16 events)
```

### In S3

If you take a look at your S3 console now, you'll see the resulting parquet files, correctly ordered by region, account id and hour in your AWS Security Lake bucket:

![](images/aws_security_lake_s3.png)

## Conclusion

> We would like to thank the AWS team for thinking about Falco to join the AWS Security Lake project since the beginning and for their precious help in the development of the integration in Falcosidekick. With this post, you should be able to send your Falco events to AWS Security Lake and run analyzes through AWS Athena. If you have any issue or request, please join us on [Slack](https://kubernetes.slack.com/?redir=%2Farchives%2FCMWH3EH32).

Reviewer: ```suggestion
We hope this step by step blog helps the AWS & Falco community get easily started using this integration. We would like to thank the AWS team for partnering with Falco and the invitation to participate in the AWS Security Lake project launch. It has been a great partnership since the beginning, we are also so appreciative of the contributions and support during the development of the integration in Falcosidekick. The goal of the post is help you get all of your configurations ready to enable the transfer of Falco events into AWS Security Lake. If you have any issues or requests, please join us on [Slack](https://kubernetes.slack.com/?redir=%2Farchives%2FCMWH3EH32).
```
